### PR TITLE
Add failing integration-test null selected fieled coalesced to zero

### PR DESCRIPTION
**Pull request by RustedBones**
Illustrate googleapis/google-cloud-java#11970 

Test failing with
```
Row not matching expectations: {"array_field": [1, 2, 3], "struct_field": {"int_field": 10, "str_field": "abc"}, "struct_field_opt": {"int_field": 0}} expected null, but was:<{"int_field": 0}>
```

### CODE REVIEW

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryStorageTest.java [hunk 1]
```diff
@@ -873,6 +883,16 @@ public void testStructAndArraySqlTypes() throws InterruptedException, IOExceptio
...
+
+    // Validate the projected STRUCT field and its members.
+    Schema unionSchema = avroSchema.getField("struct_field_opt").schema();
+    Schema structOpSchema = unionSchema.getTypes().get(1);
+    assertEquals(actualSchemaMessage, 1, structOpSchema.getFields().size());
+    assertEquals(
+            actualSchemaMessage,
+            Schema.Type.LONG,
+            structSchema.getField("int_field").schema().getType());
+    assertNull(rowAssertMessage, record.get("struct_field_opt"));
```
Comment (RustedBones) [inline google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryStorageTest.java:895]: Test fails here. whole record is `null` but returned value is `{"int_field": 0}`

**Comment by RustedBones** [comment]
As far as I see the root cause comes from the received `serialized_binary_rows`. Can the issue be escalated ?
